src/app/admin/modals/user-notification-settings-modal/notification.component.ts

```diff
@@ -0,0 +1,95 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import {
+  alertService,
+  currentUser,
+  auth,
+  User
+} from 'src/app/ajs-upgraded-providers';
+import { MatDialogRef } from '@angular/material/dialog';
+import { NgForm, FormBuilder, FormControl, FormGroup } from '@angular/forms'
+
+@Component({
+  selector: 'df-notific-setting',
+  templateUrl: './notification.component.html',
+  styleUrls: ['./user-notification-settings-modal.scss'],
+})
+export class notificationComponent implements OnInit {
+  // User: userModel;
+  feedbackControl = new FormControl();
+  portfolioControl = new FormControl();
+  taskControl = new FormControl();
+  notificationSettings: FormGroup;
+  classList: Array<string>;
+  config = {
+    animated: true, 
+    keyboard: true, 
+    backdrop: true,
+    ignoreBackdropClick: false, 
+  };
+  constructor(
+    public dialogRef: MatDialogRef<notificationComponent>,
+    fb: FormBuilder,
+    @Inject(currentUser) public CurrentUser,
+    @Inject(alertService) private alertService,
+    @Inject(auth) private auth,
+    @Inject(User) private user,
+  ) {
+    this.notificationSettings = fb.group({
+      taskSetting: new FormControl(),
+      feedbackSetting: new FormControl(),
+      portfolioSetting: new FormControl(),
+    })
+  }
+
+  ngOnInit() {
+    this.notificationSettings.setValue({
+      taskSetting: this.CurrentUser.profile.receive_task_notifications,
+      feedbackSetting: this.CurrentUser.profile.receive_feedback_notifications,
+      portfolioSetting: this.CurrentUser.profile.receive_portfolio_notifications
+    }); 
+  }
+
+  // Define the methods of changing the radio button style, to simulate styles of previous Angular/Bootstrap version
+  changeItem(val: string, judge: Boolean) {
```

Comment:
Can we rename the parameter ` val ` with a meanful word?

Reply:
Thanks for your suggestions

Reply:
changed variable names as 'setting' - indicate the group name of the setting button; 'status' - indicate the button value